### PR TITLE
feat: Allow connection to clamd through a TLS proxy

### PR DESCRIPTION
**Pull request by carboneater**
Quite a small code change, which required quite a lot of CI patches to get back in working order.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,10 +7,10 @@ on:
       - master
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [12,13,14,15,16,17]
+        node-version: [12,13,14,15,16,17,18,19]
```
Comment (carboneater) [inline .github/workflows/test.yml:13]: Node 13, 15 & 17 never made it to LTS and are EOL now. Is it still worth running tests on them?

Node 12 LTS is EOL, but I can understand why you'd want to keep testing against it

**Comment by kylefarris** [comment]
Thanks for your contribution @carboneater! Looks like the CI suite tests all got cancelled after about 1 minute. Are you able to see the results or not? If not, I'll see what I can do to get you access to seeing them.

**Comment by kylefarris** [comment]
Thanks for your contribution @carboneater! Looks like the CI suite tests all got cancelled after about 1 minute. Are you able to see the results or not? If not, I'll see what I can do to get you access to seeing them.

**Comment by carboneater** [comment]
> Thanks for your contribution @carboneater! Looks like the CI suite tests all got cancelled after about 1 minute. Are you able to see the results or not? If not, I'll see what I can do to get you access to seeing them.

Yes, I can see tests failed with
> Reason:  invalid scanner: expected promise not to be rejected with 'Error' but it was rejected with 'Error: connect ENOENT /var/run/clamav…' Error: connect ENOENT /var/run/clamav/clamd.ctl
> Unhandled Rejection reason: [AssertionError: invalid scanner: expected promise not to be rejected with 'Error' but it was rejected with 'Error: connect ENOENT /var/run/clamav…'] 
> ...

However, that's the first time I hit this case in github actions...
It's as if the same workflow ran on two very different machines...
Double-checking the workflows, when I ran the tests using `ubuntu-latest` I had `22.04` machines.
The machine running the test for Node 17 was a `20.04` machine.

According to the github post, `20.04` should be almost completely phased out of `ubuntu-latest`
So we can re-run the workflow and see how it goes.
Or we can pin `ubuntu-22.04` instead.

(Side note, even on my branch, the pipeline is flaky on the following test, but that feels like a race condition to me.)

**Comment by kylefarris** [comment]
Looks like when I re-ran the tests, everything worked. GitHub Actions don't seem super reliable.

File: README.md [hunk 2]
```diff
@@ -131,6 +131,7 @@ const ClamScan = new NodeClam().init({
         reloadDb: false, // If true, will re-load the DB on every call (slow)
         active: true, // If true, this module will consider using the clamdscan binary
         bypassTest: false, // Check to see if socket is available when applicable
+        tls: false // If true, will connect to clamd over TLS instead of plaintext TCP. (Only checked when clamdscan.socket is falsey)
```
Comment (kylefarris) [inline README.md:134]:
```suggestion
        tls: false, // Use plaintext TCP to connect to clamd
```

File: index.js [hunk 3]
```diff
@@ -104,6 +106,7 @@ class NodeClam {
      * @param {boolean} [options.clamdscan.reloadDb=false] - If true, will re-load the DB on ever call (slow)
      * @param {boolean} [options.clamdscan.active=true] - If true, this module will consider using the `clamdscan` binary
      * @param {boolean} [options.clamdscan.bypassTest=false] - If true, check to see if socket is avaliable
+     * @param {boolean} [options.clamdscan.tls=false] - If true, connect to a TLS-Termination proxy in from of ClamAV
```
Comment (kylefarris) [inline index.js:109]: You have a typo in this JSDoc comment. Did you mean "...in _form_ of ClamAV"?

File: package.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "SaltwaterC <Ștefan Rusu>",
     "Sjord <Sjoerd Langkemper>",
     "chris-maclean <Christopher Maclean>",
-    "ngraef <Nick Graef"
+    "ngraef <Nick Graef>"
```
Comment (kylefarris) [inline package.json:14]:
```suggestion
    "ngraef <Nick Graef>",
    "caroneater <Gabriel Fournier>"
```

**Review by kylefarris — CHANGES_REQUESTED** [review]
I really appreciate you taking this issue on @carboneater. Looks like dependencies needed some major updates too, so, thanks for taking care of that. I added you as a contributor in the package file :)

There's just a couple really minor things I'd like you to address and I'll get this rebased and merged in.

File: index.js (hunk 3 above)
Comment (carboneater) [reply index.js:109]: Yikes!
I was going for `in front of`

Comment (kylefarris) [reply index.js]: That makes a lot more sense, haha.

**Review by kylefarris — APPROVED** [review]

**Comment by kylefarris** [comment]
I still need to re-run the tests multiple times to get them to complete. Not sure what's up with that. But, the tests have passed so I'll get this merged in.

**Pull request merged**